explainability-service/src/main/java/org/kie/trustyai/service/payloads/ReconciledMetricRequest.java

```diff
@@ -0,0 +1,165 @@
+import javax.enterprise.inject.Instance;
+
+import org.kie.trustyai.service.data.DataSource;
+import org.kie.trustyai.service.data.metadata.Metadata;
+import org.kie.trustyai.service.payloads.values.DataType;
+import org.kie.trustyai.service.payloads.values.TypedValue;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+
+@JsonPropertyOrder({ "protected", "favorable" })
+public class ReconciledMetricRequest {
+    private String protectedAttribute;
+    private String outcomeName;
+    private String modelId;
+
+    // this is the unique name of this specific request
+    private String requestName;
+
+    // this is the name of the metric that this request calculates, e.g., DIR or SPD
+    private String metricName;
+    private Double thresholdDelta;
+
+    private Integer batchSize;
+
+    private TypedValue privilegedAttribute;
+    private TypedValue favorableOutcome;
+    private TypedValue unprivilegedAttribute;
+
+    public ReconciledMetricRequest() {
+    };
+
+    public ReconciledMetricRequest(String protectedAttribute, String outcomeName, String modelId, String requestName, String metricName, Double thresholdDelta, Integer batchSize,
+            TypedValue privilegedAttribute, TypedValue favorableOutcome, TypedValue unprivilegedAttribute) {
+        this.protectedAttribute = protectedAttribute;
+        this.outcomeName = outcomeName;
+        this.modelId = modelId;
+        this.requestName = requestName;
+        this.metricName = metricName;
+        this.thresholdDelta = thresholdDelta;
+        this.batchSize = batchSize;
+        this.privilegedAttribute = privilegedAttribute;
+        this.favorableOutcome = favorableOutcome;
+        this.unprivilegedAttribute = unprivilegedAttribute;
+    }
+
+    public ReconciledMetricRequest(BaseMetricRequest request, DataType protectedType, DataType outcomeType) {
+
+        protectedAttribute = request.getProtectedAttribute();
+        outcomeName = request.getOutcomeName();
+        modelId = request.getModelId();
+        requestName = request.getRequestName();
+        metricName = request.getMetricName();
+        thresholdDelta = request.getThresholdDelta();
+        batchSize = request.getBatchSize();
+
+        this.privilegedAttribute = new TypedValue();
+        this.privilegedAttribute.setType(protectedType);
+        this.privilegedAttribute.setValue(request.getPrivilegedAttribute());
+
+        this.unprivilegedAttribute = new TypedValue();
+        this.unprivilegedAttribute.setType(protectedType);
+        this.unprivilegedAttribute.setValue(request.getUnprivilegedAttribute());
+
+        this.favorableOutcome = new TypedValue();
+        this.favorableOutcome.setType(outcomeType);
+        this.favorableOutcome.setValue(request.getFavorableOutcome());
+    }
+
+    public static ReconciledMetricRequest reconcile(BaseMetricRequest request, Instance<DataSource> dataSource) {
+        final Metadata metadata = dataSource.get().getMetadata(request.getModelId());
```

Comment:
It is a minor comment but I would prefer to do the `get()` outside this class (aka remove this overload and keep the other method) to avoid to bring usage of `javax.enterprise.inject.Instance` (aka J2EE) into this class.
As alternative you can replace `Instance` with [`Supplier`](https://docs.oracle.com/javase/8/docs/api/java/util/function/Supplier.html) and invoke this method like `reconcile(request, dataSource::get)`
Btw this is minor, it is mainly to limit the spreading of J2EE classes